src/adapter/stream.m

```diff
@@ -328,21 +330,7 @@ extern void adapter_stream() {
 extern void adapter_stream_env() { adapter_stream(); }
 
 extern void _adapter_stream_cancel() {
-    if (g_runLoop) {
-        CFRunLoopStop(g_runLoop);
-        g_runLoop = NULL;
-    }
-}
-
-static void handleSignal(int signal) {
-    if (signal == SIGTERM) {
-        _adapter_stream_cancel();
-    }
-}
-
-__attribute__((constructor)) static void init() {
-    g_runLoop = CFRunLoopGetCurrent();
-    signal(SIGTERM, handleSignal);
```

Comment:
I don't see a reason in moving this out of init and removing init either, especially while teardown still exists.